static/js/util/redux.js

```diff
@@ -0,0 +1,25 @@
+// @flow
+import type {
+  ActionCreator,
+  AsyncActionHelper,
+  AsyncActionCreator
+} from '../flow/reduxTypes';
+import type { Dispatch } from 'redux';
+
+export function createActionHelper(dispatch: Dispatch, actionCreator: Function): (...args: any) => void {
+  return (...args) => dispatch(actionCreator(...args));
+}
+
+export type ActionHelpers = Array<{[k: string]: (...args: any) => void}>;
+export function createSimpleActionHelpers(dispatch: Dispatch, actionManifest: ActionCreator[]): ActionHelpers {
+  return actionManifest.map(actionCreator => (
+    { [actionCreator.name]: createActionHelper(dispatch, actionCreator) }
+  ));
+}
+
+export type AsyncActionHelpers = Array<{[k: string]: AsyncActionHelper}>;
+export function createAsyncActionHelpers(dispatch: Dispatch, actionManifest: AsyncActionCreator[]): AsyncActionHelpers {
+  return actionManifest.map(actionCreator => (
+    { [actionCreator.name]: createActionHelper(dispatch, actionCreator) }
+  ));
+}
```

Comment:
they're differentiated because they have different return types, `createSimpleActionHelpers` returns this:

```js
Array<{[k: string]: (...args: any) => void}>
```

an array of objects with a string key and a function that takes some arguments and returns `undefined`. these are the simple synchronous action creators that just take a value and dispatch a change to the store.

`createAsyncActionHelpers` has the following return type:

```js
Array<{[k: string]: (...a: any) => Promise}>
```

so an array of objects mapping a string to a function that takes some arguments and returns a promise. these are the async action creators.

note: we're using the [redux-thunk](https://open.spotify.com/album/16i5KnBjWgUtwOO7sVMnJB) middleware for asynchronous actions.